bats/remotes-file-system.bats

```diff
@@ -170,29 +169,6 @@ SQL
     [ "$status" -eq 0 ]
     [[ "${lines[0]}" =~ "master" ]] || false
     [[ "${lines[1]}" != "tester" ]] || false
-
-    # fetch branches from remote
-    dolt fetch
-
-    # should display both branches fetched from remote
-    run dolt branch
-    [ "$status" -eq 0 ]
-    [[ "${lines[0]}" =~ "master" ]] || false
-    skip "Fetched branch not displayed in branch list" [[ "${lines[1]}" =~ "tester" ]] || false
-
-    # should not be able to create branch with same
-    # name as remote branch
-    run dolt checkout -b tester
-    [ "$status" -eq 1 ]
-    [[ "${lines[0]}" =~ "fatal: A branch named 'tester' already exists." ]] || false
-
-    # delete tester branch
-    dolt branch -d -f tester
-    
-    # should not be able to checkout branch that is not displayed
-    run dolt checkout tester
-    [ "$status" -eq 1 ]
-    [[ "${lines[0]}" =~ "error: could not find tester" ]] || false
```

Comment:
@timsehn I think this all matches git:

```
~/dev/dolt/go>git branch -v -a | grep diff-hist-opt
  bh/diff-hist-opt                                                     dd1723442 pr feedback
  remotes/origin/bh/diff-hist-opt                                      dd1723442 pr feedback

~/dev/dolt/go>git branch -D bh/diff-hist-opt
Deleted branch bh/diff-hist-opt (was dd1723442).

~/dev/dolt/go>git fetch

~/dev/dolt/go>git branch -v -a | grep diff-hist-opt
  remotes/origin/bh/diff-hist-opt                                      dd1723442 pr feedback

~/dev/dolt/go>git checkout -b bh/diff-hist-opt
Switched to a new branch 'bh/diff-hist-opt'

~/dev/dolt/go>git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

~/dev/dolt/go>git branch -D bh/diff-hist-opt
Deleted branch bh/diff-hist-opt (was 770d05447).

~/dev/dolt/go>git checkout bh/diff-hist-opt
Branch 'bh/diff-hist-opt' set up to track remote branch 'bh/diff-hist-opt' from 'origin'.
Switched to a new branch 'bh/diff-hist-opt'
```

Reply:
if you delete a branch and fetch it you will only see the remotely tracked version of that branch, not local branch.  You can create a branch whose name exactly matches a remotely tracked branch.  If you delete a branch and then check a branch out with the same name it will check out the remotely tracked branch with that name.

Reply:
can we assert this behavior by: 
```
dolt checkout -b tester
dolt sql -q 'insert into test values (0,0,0,0,0,0)'
dolt add test
dolt commit -m "added row"
run dolt push  # don't specify upstream
[ "$status" -eq 0 ]
```